packages/salesforcedx-vscode-apex/src/commands/apexActionController.ts

```diff
@@ -132,4 +132,173 @@ export class ApexActionController {
       .join('\n');
     return stringify(parse(theDoc));
   }
+
+  private saveOasAsErsMetadata = async (oasSpec: string, fullPath: string): Promise<void> => {
+    // Replace the schema section in the ESR file if it already exists
+    let existingContent;
+    if (fs.existsSync(fullPath)) {
+      existingContent = fs.readFileSync(fullPath, 'utf8');
+    }
+    const namedCredential = await this.showNamedCredentialsQuickPick();
+
+    const updatedContent = this.buildESRXml(existingContent, fullPath, namedCredential, oasSpec);
+    try {
+      // Step 3: Write File
+      fs.writeFileSync(fullPath, updatedContent);
+      await vscode.workspace.openTextDocument(fullPath).then((newDocument: vscode.TextDocument) => {
+        void vscode.window.showTextDocument(newDocument);
+      });
+    } catch (error) {
+      throw new Error(nls.localize('artifact_failed', error.message));
+    }
+  };
+
+  private pathExists = async (filename: string): Promise<string> => {
+    // Step 1: Prompt for Folder
+    const folder = await this.getFolderForArtifact();
+    if (!folder) {
+      throw new Error(nls.localize('no_folder_selected'));
+    }
+
+    // Step 2: Verify folder exists and if not create it
+    if (!fs.existsSync(folder)) {
+      fs.mkdirSync(folder);
+    }
+
+    // Step 2: Check if File Exists
+    const fullPath = path.join(folder, filename);
+    if (fs.existsSync(fullPath)) {
+      const shouldOverwrite = await this.confirmOverwrite();
+      if (!shouldOverwrite) {
+        throw new Error(nls.localize('operation_cancelled'));
+      }
+    }
+    return fullPath;
+  };
+
+  private confirmOverwrite = async (): Promise<boolean> => {
+    const response = await vscode.window.showWarningMessage(
+      nls.localize('file_exists'),
+      { modal: true },
+      nls.localize('overwrite'),
+      nls.localize('cancel')
+    );
+    return response === nls.localize('overwrite');
+  };
+
+  private getFolderForArtifact = async (): Promise<string | undefined> => {
+    const registryAccess = new RegistryAccess();
+
+    const defaultESRFolder = path.join(
+      workspaceUtils.getRootWorkspacePath(),
+      'force-app',
+      'main',
+      'default',
+      registryAccess.getTypeByName('ExternalServiceRegistration').directoryName
+    );
+    const folderUri = await vscode.window.showInputBox({
+      prompt: nls.localize('enter_esr_path'),
+      value: defaultESRFolder
+    });
+
+    return folderUri ? path.resolve(folderUri) : undefined;
+  };
+  private showNamedCredentialsQuickPick = async (): Promise<string | undefined> => {
```

Comment:
Should we consider the use case where the named cred only exist in the workspace?

We can follow up with a new work item to handle finding named creds in the workspace.